### DISCUSSION TITLE
Iterate through orderer_json

### DISCUSSION THREAD

**Discussion by HamletDuFromage** [nlohmann/json#2628] [Q&A] [answered]
Hello, I'm trying to iterate through an ordered_json but can't figure it out. My json looks like so : 

```json
{
    "Firmware 11.0.1": "/Firmware%2011.0.1.zip",
    "Firmware 11.0.0": "/Firmware%2011.0.0.zip",
    "Firmware 10.2.0": "/Firmware%2010.2.0.zip",
    "Firmware 10.1.0": "/Firmware%2010.1.0.zip",
    "Firmware 10.0.4": "/Firmware%2010.0.4.zip",
    "Firmware 10.0.3": "/Firmware%2010.0.3.zip",
    "Firmware 10.0.2": "/Firmware%2010.0.2.zip",
    "Firmware 10.0.1": "/Firmware%2010.0.1.zip",
    "Firmware 10.0.0": "/Firmware%2010.0.0.zip",
    "Firmware 9.2.0": "/Firmware%209.2.0.zip",
    "Firmware 2.2.0": "/Firmware%202.2.0.zip",
    "Firmware 2.1.0": "/Firmware%202.1.0.zip",
    "Firmware 2.0.0": "/Firmware%202.0.0.zip",
    "Firmware 1.0.0": "/Firmware%201.0.0.zip"
}
```

here's what I do
```C++
json firmwares = nlohmann::ordered_json::parse(json_str);
for (auto it = firmwares.begin(); it != firmwares.end(); ++it) {
    std::cout << it.key() << " | " <<  it.value() << std::endl;
}
```
but it orders the keys alphabetically. Is there a way to preserve the order when iterating?

**Comment by HamletDuFromage** [comment, marked as answer]
nevermind, of course I need to declare `nlohmann::ordered_json firmwares` instead of `nlohmann::json firmwares`

**Reply by gregmarr** [reply]
Good place to use `auto`.  @nlohmann Does the "explicit" work make the conversion between `ordered_json` and `json` explicit too?

**Reply by nlohmann** [reply]
That conversion is done by

```cpp
    template < typename BasicJsonType,
               detail::enable_if_t <
                   detail::is_basic_json<BasicJsonType>::value&& !std::is_same<basic_json, BasicJsonType>::value, int > = 0 >
    basic_json(const BasicJsonType& val)
```

and is currently not excluded by `JSON_USE_IMPLICIT_CONVERSIONS`. What do you think - should this (breaking) change be added?

**Reply by gregmarr** [reply]
I would think that it would make sense to be part of JSON_USE_IMPLICIT_CONVERSIONS as long as the regular copy constructor is unaffected, which it looks like `!is_same` handles.

**Reply by gregmarr** [reply]
https://github.com/nlohmann/json/issues/2649 just reported this behavior as a bug.
